src/Databases/DatabasesCommon.cpp

```diff
@@ -258,6 +288,19 @@ StoragePtr DatabaseWithOwnTablesBase::detachTableUnlocked(const String & table_n
         throw Exception(ErrorCodes::UNKNOWN_TABLE, "Table {}.{} doesn't exist",
                         backQuote(database_name), backQuote(table_name));
     res = it->second;
+
+    SnapshotDetachedTable snapshot_detached_table;
+    snapshot_detached_table.database = it->second->getStorageID().getDatabaseName();
+    snapshot_detached_table.table = it->first;
+    if (it->second->getStorageID().hasUUID())
+    {
+        snapshot_detached_table.uuid = it->second->getStorageID().uuid;
+    }
+    snapshot_detached_table.is_permanently = false;
+    snapshot_detached_table.metadata_path = getObjectMetadataPath(snapshot_detached_table.table);
+
+    snapshot_detached_tables.emplace(it->first, std::move(snapshot_detached_table));
```

Comment:
I think its a good idea but I suggest to use designated initializers instead ctor, static func ... becase we have enough count of params and it will be grow.